tdlib: 1.6.0 -> 1.6.9

matthew-piziak opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

This dependency is required for the current version of the Emacs package `telega`. 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

matthew-piziak committed:
tdlib: 1.6.0 -> 1.6.9

fix hash

version 1.6.9 is not released, so use revision hash

matthew-piziak commented:
I'm having trouble testing this. I'm trying to allow the `telega` client to be run on NixOS. I run  `tdlib = fetchWithConfig "https://github.com/NixOS/nixpkgs/archive/pull/101948/head.tar.gz"` and then I install `pkgs.tdlib.emacsPackagesNg.telega` to hopefully pull in the updated dependency. This gives me the following stack trace:
```
error: while evaluating the attribute 'activationScript' of the derivation 'nixos-system-mus-20.09.1469.13d0c311e3a' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/activation/top-level.nix:95:5:
while evaluating the attribute 'system.activationScripts.script' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/activation/activation-script.nix:68:9:
while evaluating 'textClosureMap' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/strings-with-deps.nix:70:35, called from /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/activation/activation-script.nix:89:18:
while evaluating 'id' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/trivial.nix:14:5, called from undefined position:
while evaluating the attribute 'text' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/activation/activation-script.nix:9:5:
while evaluating the attribute 'text' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/strings-with-deps.nix:77:38:
while evaluating the attribute 'sources' of the derivation 'etc' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/etc/etc.nix:12:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/system/etc/etc.nix:20:20, called from undefined position:
while evaluating the attribute 'source' at undefined position:
while evaluating 'g' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/attrsets.nix:276:19, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:98:72, called from /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/attrsets.nix:279:20:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:465:9:
while evaluating the option `environment.etc.crontab.source':
while evaluating the attribute 'mergedValue' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:497:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:499:17, called from /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:499:12:
while evaluating 'check' at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/types.nix:256:15, called from /nix/var/nix/profiles/per-user/root/channels/nixpkgs/lib/modules.nix:499:22:
while evaluating the attribute 'allFiles' of the derivation 'crontabs' at /nix/store/3pxdiif5f4qams8ss18sq7i7728j03y1-nixpkgs-20.09.1469.13d0c311e3a/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'text' of the derivation 'system-crontab' at /nix/store/3pxdiif5f4qams8ss18sq7i7728j03y1-nixpkgs-20.09.1469.13d0c311e3a/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'passAsFile' of the derivation 'system-path' at /nix/store/3pxdiif5f4qams8ss18sq7i7728j03y1-nixpkgs-20.09.1469.13d0c311e3a/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'deps' of the derivation 'emacs-gcc-with-packages-20201026.0' at /nix/store/3pxdiif5f4qams8ss18sq7i7728j03y1-nixpkgs-20.09.1469.13d0c311e3a/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'explicitRequires' of the derivation 'emacs-packages-deps' at /nix/store/3pxdiif5f4qams8ss18sq7i7728j03y1-nixpkgs-20.09.1469.13d0c311e3a/nixpkgs/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'buildInputs' of the derivation 'emacs-telega-20200814.1652' at /nix/store/gwcmrldyqjb1pcj2w2msi61xwmy3mh3p-source/pkgs/build-support/emacs/generic.nix:30:3:
cannot coerce a set to a string, at /nix/store/gwcmrldyqjb1pcj2w2msi61xwmy3mh3p-source/pkgs/build-support/emacs/generic.nix:30:3
```

makefu commented:
@GrahamcOfBorg build tdlib

SuperSandro2000 commented on pkgs/development/libraries/tdlib/default.nix:
```diff
@@ -1,14 +1,14 @@
     repo = "td";
     rev = "v${version}";
-    sha256 = "0zlzpl6fgszg18kwycyyyrnkm255dvc6fkq0b0y32m5wvwwl36cv";
+    sha256 = "32f2338bd199dd06a1b4b5f1ad14f2d4f2868f01";
```
The has is wrong.

matthew-piziak replied:
Thank you! Fixed.

makefu commented:
```
error: while querying the derivation named 'tdlib-1.6.9':                                                                                                                 
while evaluating the attribute 'src' of the derivation 'tdlib-1.6.9' at /home/makefu/.cache/nixpkgs-review/pr-101948/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:192:1
1:                                                                                                                                                                        
hash '32f2338bd199dd06a1b4b5f1ad14f2d4f2868f01' has wrong length for hash type 'sha256'                                                                                   
```

matthew-piziak commented:
`nix-shell -p nix-review --run 'nix-review pr 101948'` succeeded:
```
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/101948/head:refs/nixpkgs-review/1
From https://github.com/NixOS/nixpkgs
 + a0140addeb1...8eeaad9129e refs/pull/101948/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /root/.cache/nixpkgs-review/pr-101948-3/nixpkgs 2b06415ca1854e1f8993a7a4e760ef88e1c98a7b
Preparing worktree (detached HEAD 2b06415ca18)
Updating files: 100% (23066/23066), done.
HEAD is now at 2b06415ca18 Merge pull request #101370 from m1cr0man/ssl-test-certs
$ nix-env -f /root/.cache/nixpkgs-review/pr-101948-3/nixpkgs -qaP --xml --out-path --show-trac
e$ git merge --no-commit 8eeaad9129e265d36c7879765865793e66eee9a0
Automatic merge went well; stopped before committing as requested
$ nix-env -f /root/.cache/nixpkgs-review/pr-101948-3/nixpkgs -qaP --xml --out-path --show-trace --meta
1 package updated:
tdlib (1.6.0 → 1.6.9)

$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /root/.cache/nixpkgs-review/pr-101948-3/build.nix
https://github.com/NixOS/nixpkgs/pull/101948
1 package built:
tdlib

$ nix-shell /root/.cache/nixpkgs-review/pr-101948-3/shell.nix
```

SuperSandro2000 commented:
Result of `nixpkgs-review pr 101948` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>haskellPackages.tdlib</li>
    <li>tdlib</li>
  </ul>
</details>

makefu commented:
maybe you could add a comment to the rev why we are not using v${version} anymore

matthew-piziak committed:
tdlib: 1.6.0 -> 1.6.9

fix hash

version 1.6.9 is not released, so use revision hash

add a comment to the rev about why we are not using `v${version}` anymore

matthew-piziak commented:
Good idea @makefu. Done.

matthew-piziak commented:
@GrahamcOfBorg build tdlib

SuperSandro2000 commented:
> @GrahamcOfBorg build tdlib

You don't have permission to do that.

matthew-piziak commented:
@SuperSandro2000 Oops, my mistake. Monkey see, monkey do.

veprbl commented on pkgs/development/libraries/tdlib/default.nix:
```diff
@@ -1,14 +1,20 @@
 
 stdenv.mkDerivation rec {
-  version = "1.6.0";
+  version = "1.6.9";
```
Since there is no tag and https://github.com/tdlib/td/commit/32f2338bd199dd06a1b4b5f1ad14f2d4f2868f01 is just a random commit
```suggestion
  version = "unstable-2020-10-25";
```

matthew-piziak committed:
tdlib: 1.6.0 -> 1.6.9

fix hash

version 1.6.9 is not released, so use revision hash

add a comment to the rev about why we are not using `v${version}` anymore

use unstable-2020-10-25

matthew-piziak commented:
Thanks @veprbl! Fixed.

veprbl commented:
The `rev` would need to stay the same, the suggestion was for the `version`.

matthew-piziak committed:
tdlib: 1.6.0 -> 1.6.9

fix hash

version 1.6.9 is not released, so use revision hash

add a comment to the rev about why we are not using `v${version}` anymore

use unstable-2020-10-25

update version, not revision

matthew-piziak commented:
@veprbl Oops, my mistake. Fixed.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-in-distress/3604/36

veprbl commented:
@GrahamcOfBorg build haskellPackages.tdlib

veprbl reviewed (approved):
LGTM, builds on x86_64-darwin

matthew-piziak commented:
/marvin opt-in

matthew-piziak commented:
/status needs_merger

matthew-piziak commented:
Strange, thought it counted as reviewed.

matthew-piziak commented:
/status awaiting_reviewer

matthew-piziak commented:
/status needs_reviewer

matthew-piziak commented:
/status needs_merger

matthew-piziak commented:
Thank you @Mic92!!
